### PR TITLE
Fixed #29502 -- Allowed autocomplete_fields to function with only view permission to related object

### PR DESCRIPTION
**Pull request by carltongibson**
https://code.djangoproject.com/ticket/29502

### CODE REVIEW

**Comment by carltongibson** [comment]
@codingjoe Could you comment? Ta. 

I wondered about having the autocomplete widget provide a better error message than "The results could not be loaded." when we get a 403 — but I didn't look at the JS yet. What do you think?

File: docs/ref/contrib/admin/index.txt [hunk 1]
```diff
@@ -1115,7 +1115,9 @@ subclass::
     more user-friendly if the related model has many instances.
 
     You must define :attr:`~ModelAdmin.search_fields` on the related object's
-    ``ModelAdmin`` because the autocomplete search uses it.
+    ``ModelAdmin`` because the autocomplete search uses it. (In addtion, this
```
Comment (timgraham) [inline docs/ref/contrib/admin/index.txt]: I would drop the parentheses, "this means that", and "in order to use autocomplete".

File: docs/ref/contrib/admin/index.txt [hunk 2]
```diff
@@ -1115,7 +1115,9 @@ subclass::
...
+    means that users are required to have ``view`` or ``change`` permissions to
```
Comment (timgraham) [inline docs/ref/contrib/admin/index.txt]: permissions -> permission
(rule: when using "A or B", both the subsequent verb and adjective must match the plurality of B.)

File: tests/admin_views/test_autocomplete_view.py [hunk 3]
```diff
@@ -81,15 +81,16 @@ def test_has_change_permission_required(self):
...
-        p = Permission.objects.get(
-            content_type=ContentType.objects.get_for_model(Question),
-            codename='change_question',
-        )
-        self.user.user_permissions.add(p)
-        request.user = User.objects.get(pk=self.user.pk)
-        response = AutocompleteJsonView.as_view(**self.as_view_args)(request)
-        self.assertEqual(response.status_code, 200)
+        for permission in ('view', 'change'):
+            self.user.user_permissions.clear()
```
Comment (timgraham) [inline tests/admin_views/test_autocomplete_view.py]: use `subTest()`

File: docs/ref/contrib/admin/index.txt (hunk 1 above)
Comment (timgraham) [reply docs/ref/contrib/admin/index.txt]: addtion -> addition

**Comment by timgraham** [comment]
Can you check if `raw_id_fields` also works with the view permission?

**Comment by carltongibson** [comment]
Thanks for making the edits @timgraham. 

> Can you check if raw_id_fields also works with the view permission?

Yes. It works perfectly. 

* No permission: 403 on the pop-up, but if I happen to know the ID creation works as expected. 
* Only view permission: Pop-up functions as expected. (I can select the related object as needed.)

**Comment by codingjoe** [comment]
@carltongibson 
> I wondered about having the autocomplete widget provide a better error message than "The results could not be loaded." when we get a 403 — but I didn't look at the JS yet. What do you think?

I guess that is possible with a [custom transport](https://select2.org/data-sources/ajax#alternative-transport-methods) you can define an a custom error handler and provide a better message. You will need to add your own translations tho :/

**Pull request merged**